packages/@azure/servicebus/data-plane/lib/receiver.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as Long from "long";
+import * as log from "./log";
+import { StreamingReceiver, MessageHandlerOptions } from "./core/streamingReceiver";
+import { BatchingReceiver } from "./core/batchingReceiver";
+import { ReceiveOptions, OnError, OnMessage, ReceiverType } from "./core/messageReceiver";
+import { ClientEntityContext } from "./clientEntityContext";
+import { ServiceBusMessage, ReceiveMode } from "./serviceBusMessage";
+
+/**
+ * Describes the options for creating a Receiver.
+ */
+export interface MessageReceiverOptions {
+  /**
+   * An enum indicating the mode in which messages should be received.
+   * Possible values are `ReceiveMode.peekLock` (default) and `ReceiveMode.receiveAndDelete`
+   */
+  receiveMode?: ReceiveMode;
+}
+
+/**
+ * An abstraction over the underlying receiver link.
+ * The Reciever class can be used to recieve messages in a batch or by registering handlers
+ */
```

Comment:
Would update documentation here to be clear about running multiple receivers and caveats involved.